### ISSUE TITLE
How to calculate ecs task run

### ISSUE THREAD

**Issue by joanayma** [infracost/infracost#1300]
Hi,

We are using AWS Step Functions that has some steps for lambdas and some (quite a lot actually) are ecs tasks. We have the task definitions `aws_ecs_task_definitions`, which makes sense that are free, and `aws_sfn_state_machine` in the same terraform code. The SFN has transactions usage number, but the ecs task is out of scope. While ecs task definition is free, running the task isn't.

How we can add cost estimations (hours) to those resources? I'm willing to implement this but it's unclear to what add that usage estimation.

Thanks.

**Comment by aliscott** [comment]
@joanayma thanks! We have a [usage file](https://www.infracost.io/docs/features/usage_based_resources/#infracost-usage-file) that can be used to specify usage-based properties for a resource, so perhaps we could utilize that when adding support for this?

How would you think this could be modelled? One idea I have is to parse the `container_definitions` to sum up the CPU and memory numbers, and then we can multiply that by the hours that are specified in the usage file. But I'm not sure how this works for both EC2 and Fargate.

**Comment by joanayma** [comment]
Yes, parsing the container_definitions is definitely the way to go. Then it requires the (multiple) region where the user is going to use it. As far I see, the _add new resources_ contribution readme shows something about how that can be done. And for EC2 this is similar or using what has done in #79.

Right now ecs task definition is in the _free resources_ list. Should we move it as a new resource? As stated, the task definitions are free and if not moved out of the list AFAIK we can't add estimations for the usage file.

**Comment by aliscott** [comment]
Yeah adding it as a new resource, and moving it out of the free resources makes sense.

For the EC2 option would we need some way of working out the instance count of the instances that would be launched? Or would it be enough for us to specify that in the usage file?

**Comment by aliscott** [comment]
@joanayma I was looking at prioritizing this and wondered if you had any thoughts on my question above?

**Comment by joanayma** [comment]
Hi sorry too busy week. I think the best is usage-file for things that are not in containers_definitions. Also I think the task run only runs one instance, not sure, but definitely a "count" of how many times the user is going to run or a total of running hours is a user input.

**Comment by aliscott** [comment]
Thanks! Yeah I think total running hours makes a lot sense.

**Comment by remidebette** [comment]
Hi, 
Thanks for the nice tool.

This would be very convenient for us since ECS Tasks are our main source of compute.
Would you have any ETA?

**Comment by alikhajeh1** [comment]
@remidebette thanks! Unfortunately we don't have an ETA. The above chat between @aliscott and @joanayma seem to have outlined the required work, we'd love contributions if anyone in the community would like to help.

**Comment by anitakrueger** [comment]
While most of our task definitions are run in ECS Service, we have a few that are being scheduled by EventBridge so would need to be usage based as well. Would love to see this implemented.

**Comment by dobeerman** [comment]
Any updates on this? We also have numerous task definitions and can't use the tool since it doesn't include task definitions in the estimation.

**Comment by glenngillen** [comment]
We're moving to a contribution model where Issues will be exclusively for work that is ready to be worked on. Discussions and feature requests about potential future capabilities will be over in [Discussions](https://github.com/infracost/infracost/discussions). There has been very little community or customer interest in this capability for an extended period, but if it still interests you (or you're now reading this issue because you've been looking for a solution) please [create a new thread in Discussions](https://github.com/infracost/infracost/discussions).
